x-pack/solutions/security/plugins/security_solution/public/flyout/document_details/right/footer.tsx

```diff
@@ -40,6 +50,18 @@ export const PanelFooter: FC<PanelFooterProps> = ({ isRulePreview }) => {
     dataFormattedForFieldBrowser,
     isAlert,
   });
+  const isAgentBuilderEnabled = useIsExperimentalFeatureEnabled('agentBuilderEnabled');
+
+  const alertData = useMemo(() => {
+    const rawData = getRawData(dataFormattedForFieldBrowser ?? []);
+    return stringifyEssentialAlertData(rawData);
+  }, [dataFormattedForFieldBrowser]);
+
+  const { openAgentBuilderFlyout } = useAgentBuilderAttachment({
+    attachmentType: SecurityAgentBuilderAttachments.alert,
+    attachmentData: { alert: alertData },
+    attachmentPrompt: isAlert ? ALERT_ATTACHMENT_PROMPT : EVENT_ATTACHMENT_PROMPT,
+  });
```

Comment:
Unless I'm mistaken, this piece of code is duplicated between here and the ease flyout. What do you think about moving this into a hook that would be shared between the 2 flyouts?

I'm bringing this up because we will also need this code again in the new attacks flyout (see the footer [here](https://github.com/elastic/kibana/blob/main/x-pack/solutions/security/plugins/security_solution/public/flyout/attack_details/footer.tsx)). At this time we do not have the `Ask AI assistant` button there, but it's coming [very soon](https://github.com/elastic/kibana/issues/237374) (cc @agusruidiazgd)

Reply:
They have different prompt arguments. I think its fine to call `useAgentBuilderAttachment` directly, but if you want to create a hook when the time comes feel free